From 3fb282977fa4aeb1d23a8a07b4ebd89d804db059 Mon Sep 17 00:00:00 2001
From: Vivek Natarajan <nataraja@codeaurora.org>
Date: Thu, 8 Oct 2015 12:57:51 +0530
Subject: [PATCH 257/500] Support preallocation of skbs

This patch preallocates SKBs each of 4K size in recycling lists.
The number of SKBs are configured through
CONFIG_SKB_RECYCLE_MAX_PREALLOC_SKBS.

Change-Id: I581813dfb298da2844aca86bebf9d80399d629ed
Signed-off-by: Vivek Natarajan <nataraja@codeaurora.org>
Signed-off-by: Casey Chen <kexinc@codeaurora.org>
---
 net/Kconfig | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/net/Kconfig b/net/Kconfig
index ce41802c488c..33331b4cd5d1 100644
--- a/net/Kconfig
+++ b/net/Kconfig
@@ -375,6 +375,24 @@ config SKB_RECYCLER_MULTI_CPU
 	depends on SMP && SKB_RECYCLER
 	default n
 
+config SKB_RECYCLER_PREALLOC
+	bool "Enable preallocation of SKBs"
+	depends on SKB_RECYCLER
+	default n
+	help
+	 Preallocates SKBs in recycling lists and the number of
+	 SKBs are configured through CONFIG_SKB_RECYCLE_MAX_PREALLOC_SKBS.
+	 This needs SKB_RECYCLER to be enabled.
+	 The number of preallocated SKBs can be passed using
+	 SKB_RECYCLE_MAX_PREALLOC_SKBS.
+
+config SKB_RECYCLE_MAX_PREALLOC_SKBS
+	int "Number of SKBs to be preallocated"
+	depends on SKB_RECYCLER_PREALLOC
+	default 16384
+	help
+	 Number of SKBs each of 4K size to be preallocated for recycling
+
 menu "Network testing"
 
 config NET_PKTGEN
-- 
2.34.1

